components/cloud_usage_modal/lhs_nearing_limit_modal.tsx

```diff
@@ -0,0 +1,104 @@
+
+import CloudUsageModal from './index';
+
+export default function LHSNearingLimitsModal() {
+    const dispatch = useDispatch();
+    const product = useSelector(getSubscriptionProduct);
+    const usage = useGetUsage();
+    const intl = useIntl();
+    const openPricingModal = useOpenPricingModal();
+
+    const [limits] = useGetLimits();
+
+    const primaryAction = {
+        message: {
+            id: t('workspace_limits.modals.view_plans'),
+            defaultMessage: 'View plans',
+        },
+        onClick: openPricingModal,
+    };
+    const secondaryAction = {
+        message: {
+            id: t('workspace_limits.modals.close'),
+            defaultMessage: 'Close',
+        },
+        onClick: () => {
+            dispatch(closeModal(ModalIdentifiers.CLOUD_LIMITS));
+        },
+    };
+    const highestLimit = useGetHighestThresholdCloudLimit(usage, limits);
+    let title: Message = {
+        id: t('workspace_limits.modals.informational.title'),
+        defaultMessage: '{planName} limits',
+        values: {
+            planName: product?.name,
+        },
+    };
+
+    let description: Message = {
+        id: t('workspace_limits.modals.informational.description.starterLimits'),
+        defaultMessage: '{planName} is restricted to {messages} message history, {storage} file storage, {integrations} apps, and {boards} board cards.',
+        values: {
+            planName: product?.name,
+            messages: intl.formatNumber(limits?.messages?.history ?? fallbackStarterLimits.messages.history),
+            storage: asGBString(limits?.files?.total_storage ?? fallbackStarterLimits.files.totalStorage, intl.formatNumber),
+            integrations: limits?.integrations?.enabled ?? fallbackStarterLimits.integrations.enabled,
+            boards: limits?.boards?.cards ?? fallbackStarterLimits.boards.cards,
+        },
+    };
+    if (product?.sku === CloudProducts.PROFESSIONAL) {
+        description = {
+            id: t('workspace_limits.modals.informational.description.professionalLimits'),
+            defaultMessage: '{planName} is restricted to {storage} file storage.',
+            values: {
+                planName: product.name,
+                storage: asGBString(limits?.files?.total_storage ?? fallbackStarterLimits.files.totalStorage, intl.formatNumber),
+            },
+        };
+    }
+
+    if (highestLimit && highestLimit.id === LimitTypes.messageHistory) {
+        title = {
+            id: t('workspace_limits.modals.limits_reached.title.message_history'),
+            defaultMessage: 'Message history',
+        };
+
+        description = {
+            id: t('workspace_limits.modals.limits_reached.description.message_history'),
+            defaultMessage: 'Your sent message history is no longer available but you can still send messages. Upgrade to a paid plan and get unlimited access to your message history.',
+        };
+    }
```

Comment:
Ambiguity around the design comment

> This modal should only appear for message history limit reached and not for other limits - so as to reduce interruptions to the admin.
>
> this should be flexible and something we return to if we feel we need to show it more for the other limits. To start off I’d like to limit the interruptions.

In the interest of time, it seemed ok to proceed without other overrides since the information and actions are still present.